build-tools/packages/build-cli/src/commands/release/setDtsTypes.ts

```diff
@@ -0,0 +1,182 @@
+				if (isDtsKind(input)) {
+					return input;
+				}
+				throw new Error(`Invalid release type: ${input}`);
+			},
+		})(),
+		...PackageCommand.flags,
+	};
+
+	private readonly packageList: PackageTypesList = {
+		packagesNotUpdated: [],
+		packagesUpdated: [],
+	};
+
+	protected async processPackage(pkg: Package): Promise<void> {
+		updatePackageJsonFile(pkg.directory, (json) => {
+			if (json.types !== undefined && json.typings !== undefined) {
+				throw new Error(
+					"Both 'types' and 'typings' fields are defined in the package.json. Only one should be used.",
+				);
+			}
+
+			const types: string | undefined = json.types ?? json.typings;
+
+			if (types === undefined) {
+				this.verbose("Neither 'types' nor 'typings' field is defined in the package.json.");
+				return;
+			}
+
+			const configOptions = ExtractorConfig.tryLoadForFolder({
+				startingFolder: pkg.directory,
+			});
+			if (configOptions === undefined) {
+				this.verbose(`No api-extractor config found for ${pkg.name}. Skipping.`);
+				return;
+			}
+			const extractorConfig = ExtractorConfig.prepare(configOptions);
+			assert(this.flags.types !== undefined, "--types flag must be provided.");
+
+			const packageUpdated = updatePackageJsonTypes(
+				pkg.directory,
+				extractorConfig,
+				this.flags.types,
+				json,
+				this.logger,
+			);
+
+			const updatedPackageList = packageUpdated
+				? this.packageList.packagesUpdated.push(pkg.name)
+				: this.packageList.packagesNotUpdated.push(pkg.name);
+			return updatedPackageList;
+		});
+	}
+
+	public async run(): Promise<PackageTypesList> {
+		// Calls processPackage on all packages.
+		await super.run();
+
+		if (this.packageList.packagesUpdated.length === 0) {
+			this.log(`No updates in package.json for ${this.flags.types} release tag`);
+		}
+
+		return this.packageList;
+	}
+}
+
+/**
+ * Updates the types/typing field in package.json.
+ * @returns true if the update was successful, false otherwise.
+ */
+function updatePackageJsonTypes(
+	directory: string,
+	extractorConfig: ExtractorConfig,
+	dTsType: DtsKind,
+	json: PackageJson,
+	log: CommandLogger,
+): boolean {
+	try {
+		if (extractorConfig.rollupEnabled === true) {
+			let filePath = "";
+
+			switch (dTsType) {
+				case "alpha":
+					filePath = extractorConfig.alphaTrimmedFilePath;
+					break;
+				case "beta":
+					filePath = extractorConfig.betaTrimmedFilePath;
+					break;
+				case "public":
+					filePath = extractorConfig.publicTrimmedFilePath;
+					break;
+				case "untrimmed":
+					filePath = extractorConfig.untrimmedFilePath;
+					break;
+				default:
+					log.errorLog(`${dTsType} is not a valid value.`);
+					break;
+			}
+
+			if (filePath) {
+				if (json.typings !== undefined) {
+					json.typings = calculateRelativePath(filePath, directory);
+					return true;
+				}
+				assert(json.types !== undefined, "`types` field is undefined");
+				json.types = calculateRelativePath(filePath, directory);
+				return true;
```

Comment:
Don't duplicate code like this: its bad for maintainability. If you want to preserve this logic, do something more like:
```suggestion
				const typings = calculateRelativePath(filePath, directory);
				if (json.typings !== undefined) {
					json.typings = typings;
				} else {
					assert(json.types !== undefined, "`types` field is undefined");
					json.types = typings;
				}
				return true;
```

Personally though I'd prefer just:

```suggestion
				delete json.typings;
				json.types = calculateRelativePath(filePath, directory);
				return true;
```

since I see no value in preserving which field was used, and making our package.json files more consistent is kinda nice.